### ISSUE TITLE
No rebuilding of image for dev.autoreload.deployments

### ISSUE THREAD

**Issue by DanielGlaas** [devspace-cloud/devspace#838]
**Is your feature request related to a problem?**  
When I change the deployment rule of an image in the *.yaml file, which (without devspace) I would manually apply by kubectl, I would expect that the image is not rebuild by docker but only the new deployment rules are used.

**Which solution do you suggest?**  
I suggest to exclude the building of a new image version if only the kubectl manifests are changed as the image itself is unchanged.
Whether this affects also other kinds of deployment I'm not able to say.

**Which alternative solutions exist?**  
I don't know.

**Additional context**  
The background of this feature request is that it would be nice to change the number of replicas in the kubectl yaml file without forcing the complete application to rebuild and restart. This would be nice to test load balancing on the fly.

/kind feature

**Comment by FabianKramm** [comment]
@DanielGlaas thanks for opening this issue! Usually this should work, however I suspect that the deployment yaml is in the building context of the image, could you try to create a .dockerignore that ignores the yaml file and then retry running devspace?

**Comment by DanielGlaas** [comment]
Thank you for the quick reply and the hint with the building context of the image!
I added the yaml files to the .dockerignore file but this didn't help. Instead I shrinked the build context itself by specifying a subdictory as context where the yaml files are not included anymore.

I must admit that I'm new to docker, kubernetes and devspace. Thus, maybe this was just a stupid mistake of a noob...

Nevertheless, in the [documentation of autoreload](https://devspace.cloud/docs/cli/development/configuration/auto-reloading) there it is always written

> DevSpace will stop the development mode, rebuild the images, redeploy the application and restart the devepment mode afterwards.

Reading this, I expected that rebuilding the images is the expected behaviour. I suggest to write this a bit more precisely.
